rootfs/etc/services.d/home-assistant/run

```diff
@@ -2,9 +2,11 @@
 # ==============================================================================
 # Start Home Assistant service
 # ==============================================================================
-cd /config || bashio::exit.nok "Can't find config folder!"
 
-# Enable Jemalloc for Home Assistant Core
-export LD_PRELOAD="/usr/local/lib/libjemalloc.so.2"
+cd /config || bashio::exit.nok "Can't find config folder!"
 
+# Enable Jemalloc for Home Assistant Core, unless disabled
+if [ -z "${DISABLE_JEMALLOC+x}" ]; then
```

Comment:
Defining a variable removes the need to juggle with undefined.
Adding this to the top might help:

```
declare DISABLE_JEMALLOC
```

Adjust this line too, adding double brackets, as we are using Bash here.

```suggestion
if [[ -z "${DISABLE_JEMALLOC}" ]]; then
```

Reply:
added double brackets,  however I left the syntax that deals with undefined variable.

`declare DISABLE_JEMALLOC`  still breaks with  `set -u` in bash ,  declaring with value makes script work, but defeats the point of reading value from env.   `run` script uses `bashio` shell, which sets `set -u`

example, from shell inside homeassitant container:
```
bash-5.0# cat test
#!/usr/bin/env bash
set -u
declare test_var;
if [[ -z $test_var ]];then
  echo test_var string is null
fi
bash-5.0# ./test
./test: line 4: test_var: unbound variable
```
your suggestion would work if `set -u` would not be there
```
bash-5.0# cat test
#!/usr/bin/env bash
declare test_var;
if [[ -z $test_var ]];then
  echo test_var string is null
fi
bash-5.0# ./test
test_var string is null
```